src/intersection-observer.js

```diff
@@ -81,20 +81,29 @@ export function getIntersectionChangeEntry(element, owner, viewport) {
       'Negative dimensions in element.');
   // Building an IntersectionObserverEntry.
 
-  let intersectionRect = element;
+  // The element is relative to (0, 0), while the viewport moves. So, we must
+  // adjust.
+  const boundingClientRect = moveLayoutRect(element, -viewport.left,
```

Comment:
I'm still fighting with this problem. It fixes the normal case (non-fixed), but breaks fixed.

I think I'm going to have to change how we calculate the position of fixed elements. We currently add the viewport to their bounding box, where we shouldn't -- most of the time. I was really hoping to avoid Layers.